src/core/utils/internal/parseGraphQLRequest.ts

```diff
@@ -40,7 +39,9 @@ export function parseDocumentNode(node: DocumentNode): ParsedGraphQLQuery {
   }
 }
 
-function parseQuery(query: string): ParsedGraphQLQuery | Error {
+async function parseQuery(query: string): Promise<ParsedGraphQLQuery | Error> {
+  const { parse } = await import('graphql')
```

Comment:
Import cache should help us here since every GraphQL handler calls `parseQuery()`.